libgoal/participation.go

```diff
@@ -200,9 +207,24 @@ func (c *Client) InstallParticipationKeys(inputfile string) (part account.Partic
 		return
 	}
 
-	partkey.Store = newdb
-	err = partkey.Persist()
-	return partkey, newdbpath, err
+	newpartkey := partkey
+	newpartkey.Store = newdb
+	err = newpartkey.Persist()
+	if err != nil {
+		return
+	}
+
+        // After successful install, remove the input copy of the
+        // partkey so that old keys cannot be recovered after they
+        // are used by algod.  We try to delete the data inside
+        // sqlite first, so the key material is zeroed out from
+        // disk blocks, but regardless of whether that works, we
+        // delete the input file.  The consensus protocol version
+	// is irrelevant for the maxuint64 round number we pass in.
+	partkey.DeleteOldKeys(basics.Round(math.MaxUint64), config.Consensus[protocol.ConsensusCurrentVersion])
+	os.Remove(inputfile)
+
+	return partkey, newdbpath, nil
```

Comment:
return newpartkey?

Reply:
Oops, you're right -- thanks for carefully looking it over!

Reply:
We're in production now - every review needs to be thorough...  We should be more diligent about unit tests for all new code as well, but that's another discussion.